sys/init.c

```diff
@@ -577,28 +577,49 @@ DokanGetMountPointList(__in PREQUEST_CONTEXT RequestContext) {
   PDOKAN_MOUNT_POINT_INFO dokanMountPointInfo;
   USHORT i = 0;
 
-  __try {
-    ExAcquireResourceExclusiveLite(
-        &RequestContext->DokanGlobal->MountPointListLock, TRUE);
-    dokanMountPointInfo = (PDOKAN_MOUNT_POINT_INFO)
-                              RequestContext->Irp->AssociatedIrp.SystemBuffer;
-    for (listEntry = RequestContext->DokanGlobal->MountPointList.Flink;
-         listEntry != &RequestContext->DokanGlobal->MountPointList;
-         listEntry = listEntry->Flink, ++i) {
+  BOOLEAN isAdmin =
+      SeSinglePrivilegeCheck(RtlConvertLongToLuid(SE_IMPERSONATE_PRIVILEGE),
+                             RequestContext->Irp->RequestorMode);
+
+  ULONG sessionId;
+  if (!NT_SUCCESS(IoGetRequestorSessionId(RequestContext->Irp, &sessionId))) {
+    sessionId = 0;
+  }
+
+  status = STATUS_SUCCESS;
```

Comment:
That's a valid change but maybe then directly set `STATUS_SUCCESS` at the declaration of the variable ? (instead of `STATUS_INVALID_PARAMETER`)